File: real estate web/FE/src/app/entity/customer/customer.ts
import {Account} from '../account/account';

export interface Customer {

  /**
   * Create by: HocHH
   * Date created: 03/02/2023
   * Uses: List customer
   */
  id_customer?: number;
  name_customer?: string;
  email_customer?: string;
  address_customer?: string;
  idCard_customer?: string;
  code_customer?: string;
  gender_customer?: number;
  flag_delete?: boolean;
  approval_customer?: number;
  phone_customer1?: string;
  phone_customer2?: string;
  account?: Account;
  dateOfBirth?: string;
  idCustomer?: number;
  nameCustomer?: string;
  emailCustomer?: string;
  addressCustomer?: string;
  idCardCustomer?: string;
  codeCustomer?: string;
  genderCustomer?: number;
  dateOfBirthCustomer?: string;
  flagDelete?: boolean;
  approvalCustomer?: number;
  phoneCustomer1?: string;
  phoneCustomer2?: string;
  accountCustomer?: Account;
  encryptPassword?: string;


}

File: real estate web/BE/src/main/java/com/c0722g1repobe/entity/customer/Customer.java
package com.c0722g1repobe.entity.customer;

import com.c0722g1repobe.entity.account.Account;
import com.c0722g1repobe.entity.post.Post;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "Customer", uniqueConstraints = {
        @UniqueConstraint(columnNames = {
                "idCardCustomer"
        }),
        @UniqueConstraint(columnNames = {
                "codeCustomer"
        })
})
@Builder
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idCustomer;
    private String nameCustomer;
    private String emailCustomer;
    private String addressCustomer;
    private String idCardCustomer;
    private String codeCustomer;
    private Integer genderCustomer;
    private String dateOfBirth;
    private boolean flagDelete = false;
    private int approvalCustomer;
    private String phoneCustomer1;
    private String phoneCustomer2;
    @OneToOne
    private Account account;
    @JsonManagedReference
    @OneToMany(mappedBy = "customer")
    private Set<Post> postSet;
}
